src/NServiceBus.Metrics/RawData/RawDataReporter.cs

```diff
@@ -0,0 +1,126 @@
+﻿namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using Hosting;
+    using Logging;
+    using Routing;
+    using Support;
+    using Transport;
+
+    delegate void WriteOutput(ArraySegment<RingBuffer.Entry> entries, BinaryWriter outputWriter);
+
+    class RawDataReporter
+    {
+        const int DefaultFlushSize = RingBuffer.Size / 2;
+        readonly RingBuffer buffer;
+        readonly int flushSize;
+        readonly Action<ArraySegment<RingBuffer.Entry>> outputWriter;
+        readonly IDispatchMessages dispatcher;
+        readonly UnicastAddressTag destination;
+        readonly TransportTransaction transportTransaction = new TransportTransaction();
+        readonly Dictionary<string, string> headers = new Dictionary<string, string>();
+        readonly BinaryWriter writer;
+        readonly MemoryStream memoryStream;
```

Comment:
Since the stream is owned by this class should the class become IDisposable and being disposed by the instance that uses it?